errors.go

```diff
@@ -134,6 +130,32 @@ func (e *pgconnError) Unwrap() error {
 	return e.err
 }
 
+// ErrTimeout occurs when an error was caused by a timeout. Specifically, it wraps an error which is
+// context.Canceled, context.DeadlineExceeded, or an implementer of net.Error where Timeout() is true.
+type ErrTimeout struct {
+	err error
+}
+
+func (e *ErrTimeout) Error() string {
+	return fmt.Sprintf("timeout: %s", e.err.Error())
+}
+
+func (e *ErrTimeout) SafeToRetry() bool {
+	var ctxErr *contextAlreadyDoneError
+	if errors.As(e, &ctxErr) {
+		return ctxErr.SafeToRetry()
+	}
+	var netErr net.Error
+	if errors.As(e, &netErr) {
+		return netErr.Temporary()
```

Comment:
I'm guessing this was meant this to be `netErr.Timeout`?

But actually, I think this entire function could be simplified to one line `return pgconn.SafeToRetry(e.err)`.

Reply:
Right on both accounts, changed: 648e9b0